### PR TITLE
Add csv tests in federation tests. Refactor non standard builder factory to be cleaner. Rename standardKeys variables to defaultKeys

### PR DESCRIPTION
**Pull request by julia-zack**

### CODE REVIEW

File: rskj-core/src/test/java/co/rsk/peg/NonStandardFederationsTest.java [hunk 1]
```diff
@@ -129,61 +128,172 @@ void createInvalidLegacyErpFederation_emptyErpKeys() {
...
+        assertTrue(builder instanceof NonStandardErpRedeemScriptBuilderWithCsvUnsignedBE);
+    }
+
+
+    @Test
+    void createValidNonStandardErpFederation_csvValueTwoBytesLong_post_RSKIP293() {
+        when(activations.isActive(ConsensusRule.RSKIP284)).thenReturn(true);
+        when(activations.isActive(ConsensusRule.RSKIP293)).thenReturn(true);
+
+        activationDelayValue = 500L;
```
[the remark this answers is not in the document]
Comment (jeremy-then) [reply rskj-core/src/test/java/co/rsk/peg/NonStandardFederationsTest.java]: I ran the tests with this line commented and all tests passed. Meaning that many of these assignments are not really needed.

File: rskj-core/src/test/java/co/rsk/peg/NonStandardFederationsTest.java [hunk 2]
```diff
@@ -129,61 +128,172 @@ void createInvalidLegacyErpFederation_emptyErpKeys() {
...
+    }
+
+    @Test
+    void createValidNonStandardErpFederation_csvValueTwoBytesLongIncludingSign_post_RSKIP293() {
+        when(activations.isActive(ConsensusRule.RSKIP284)).thenReturn(true);
+        when(activations.isActive(ConsensusRule.RSKIP293)).thenReturn(true);
+
+        activationDelayValue = 130; // Any value above 127 needs an extra byte to indicate the sign
+
+        createAndValidateFederation();
```
Comment (jeremy-then) [inline rskj-core/src/test/java/co/rsk/peg/NonStandardFederationsTest.java]: Why are you calling this on most tests? Does it make any difference every time it is called? Can this be called once somewhere else?

**Review by jeremy-then — COMMENTED** [review]
Left a couple of comments.

File: rskj-core/src/test/java/co/rsk/peg/NonStandardFederationsTest.java [hunk 3]
```diff
@@ -129,61 +128,172 @@ void createInvalidLegacyErpFederation_emptyErpKeys() {
...
 
     @Test
-    void createInvalidNonStandardBuilder_aboveMaxScriptSigSize() {
+    void createErpRedeemScriptWithCsvUnsignedBE_csvValueOneByteLong_post_RSKIP284_pre_RSKIP293() {
+        // should create the redeem script with NonStandardErpRedeemScriptBuilderWithCsvUnsignedBE
+        networkParameters = NetworkParameters.fromID(NetworkParameters.ID_TESTNET);
+        when(activations.isActive(ConsensusRule.RSKIP284)).thenReturn(true);
+        when(activations.isActive(ConsensusRule.RSKIP293)).thenReturn(false);
+
+        // For a value that only uses 1 byte it should add leading zeroes to complete 2 bytes
```
[the remark this answers is not in the document]
Comment (julia-zack) [reply rskj-core/src/test/java/co/rsk/peg/NonStandardFederationsTest.java:163]: the validation is being done in the createAndValidateFederation method, there it checks for the csvValue's bytes lengthto be 2

File: rskj-core/src/test/java/co/rsk/peg/NonStandardFederationsTest.java (hunk 2 above)
Comment (julia-zack) [reply rskj-core/src/test/java/co/rsk/peg/NonStandardFederationsTest.java]: is different because the federation created is different each time

**Pull request closed without merging**